### PR TITLE
unit test 2092

### PR DESCRIPTION
**Pull request by MarianDiakiv**

### CODE REVIEW

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 1]
```diff
@@ -1,102 +1,128 @@
...
+    @Test
+    void delete() {
+        HabitStatusCalendar habitStatusCalendar = ModelUtils.getHabitStatusCalendar();
+        HabitStatusCalendarVO habitStatusCalendarVO = ModelUtils.getHabitStatusCalendarVO();
+        when(modelMapper.map(habitStatusCalendarVO, HabitStatusCalendar.class))
+            .thenReturn(habitStatusCalendar);
+        doNothing().when(habitStatusCalendarRepo).delete(habitStatusCalendar);
+        habitStatusCalendarRepo.delete(habitStatusCalendar);
+        verify(habitStatusCalendarRepo, times(1)).delete(habitStatusCalendar);
+        habitStatusCalendarService.delete(habitStatusCalendarVO);
```
Comment (forestik) [inline service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java:73]: Same here like deleteAllByHabitAssign()

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 2]
```diff
@@ -1,102 +1,128 @@
...
+        when(habitStatusCalendarRepo.findHabitStatusCalendarByEnrollDateAndHabitAssign(LocalDate.now(), habitAssign))
+            .thenReturn(habitStatusCalendar);
+
+        when(modelMapper.map(habitStatusCalendar, HabitStatusCalendarVO.class)).thenReturn(habitStatusCalendarVO);
+        assertEquals(habitStatusCalendarVO, habitStatusCalendarService
+            .findHabitStatusCalendarByEnrollDateAndHabitAssign(LocalDate.now(), habitAssignVO));
+        assertEquals(null, habitStatusCalendarService.findHabitStatusCalendarByEnrollDateAndHabitAssign(LocalDate.now(),
+            new HabitAssignVO()));
+
+        // assertEquals(null,null);
```
Comment (forestik) [inline service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java:50]: Remove comment

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 3]
```diff
@@ -1,102 +1,128 @@
...
+
+    @Test
+    void delete() {
+        HabitStatusCalendar habitStatusCalendar = ModelUtils.getHabitStatusCalendar();
+        HabitStatusCalendarVO habitStatusCalendarVO = ModelUtils.getHabitStatusCalendarVO();
+        when(modelMapper.map(habitStatusCalendarVO, HabitStatusCalendar.class))
+            .thenReturn(habitStatusCalendar);
+        doNothing().when(habitStatusCalendarRepo).delete(habitStatusCalendar);
+        habitStatusCalendarRepo.delete(habitStatusCalendar);
+        verify(habitStatusCalendarRepo, times(1)).delete(habitStatusCalendar);
```
[the remark this answers is not in the document]
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java]: done

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 4]
```diff
@@ -1,102 +1,128 @@
...
+    }
+
+    @Test
+    void deleteAllByHabitAssign() {
+        HabitAssignVO habitAssignVO = ModelUtils.getHabitAssignVO();
+        HabitAssign habitAssign = ModelUtils.getHabitAssign();
+        when(modelMapper.map(habitAssignVO, HabitAssign.class)).thenReturn(habitAssign);
+        doNothing().when(habitStatusCalendarRepo).deleteAllByHabitAssign(habitAssign);
+        habitStatusCalendarRepo.deleteAllByHabitAssign(habitAssign);
+        verify(habitStatusCalendarRepo, times(1)).deleteAllByHabitAssign(habitAssign);
```
[the remark this answers is not in the document]
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java]: done

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 5]
```diff
@@ -1,102 +1,128 @@
...
+    void findEnrolledDatesBefore() {
+        HabitAssignVO habitAssignVO = ModelUtils.getHabitAssignVO();
+        HabitAssign habitAssign = ModelUtils.getHabitAssign();
+        HabitStatusCalendar habitStatusCalendar = ModelUtils.getHabitStatusCalendar();
+        List<HabitStatusCalendar> list = Collections.singletonList(habitStatusCalendar);
+        List<LocalDate> dates = new ArrayList<>();
+        when(modelMapper.map(habitAssignVO, HabitAssign.class)).thenReturn(habitAssign);
+        when(habitStatusCalendarRepo.findAllByEnrollDateBeforeAndHabitAssign(LocalDate.now(), habitAssign))
+            .thenReturn(list);
+        for (HabitStatusCalendar calendar : list) {
```
[the remark this answers is not in the document]
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java]: done

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 6]
```diff
@@ -1,102 +1,128 @@
...
+        assertEquals(dates, habitStatusCalendarService.findEnrolledDatesBefore(LocalDate.now(), habitAssignVO));
+    }
+
+    @Test
+    void deleteAllByHabitAssign() {
+        HabitAssignVO habitAssignVO = ModelUtils.getHabitAssignVO();
+        HabitAssign habitAssign = ModelUtils.getHabitAssign();
+        when(modelMapper.map(habitAssignVO, HabitAssign.class)).thenReturn(habitAssign);
+        doNothing().when(habitStatusCalendarRepo).deleteAllByHabitAssign(habitAssign);
+        habitStatusCalendarRepo.deleteAllByHabitAssign(habitAssign);
```
[the remark this answers is not in the document]
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java]: done

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java [hunk 7]
```diff
@@ -1,102 +1,128 @@
...
+        verify(habitStatusCalendarRepo, times(1)).deleteAllByHabitAssign(habitAssign);
+        habitStatusCalendarService.deleteAllByHabitAssign(habitAssignVO);
```
[the remark this answers is not in the document]
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java:116]: done

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java (hunk 2 above)
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java]: done

File: service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java (hunk 1 above)
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitStatusCalendarServiceImplTest.java:66]: done

File: service/src/test/java/greencity/service/HabitServiceImplTest.java [hunk 8]
```diff
@@ -1,77 +1,114 @@
...
+        when(habitTranslationRepo.findAllByTagsAndLanguageCode(pageable, lowerCaseTags, "en"))
+            .thenReturn(habitTranslationPage);
+        assertEquals(pageableDto, habitService.getAllByTagsAndLanguageCode(pageable, tags, "en"));
+    }
+
+    @Test
+    void getShoppingListForHabit() {
+        GoalTranslation goalTranslation = ModelUtils.getGoalTranslation();
+        List<GoalTranslation> goalTranslations = Collections.singletonList(goalTranslation);
+        GoalDto goalDto = modelMapper.map(goalTranslations, GoalDto.class);
```
[the remark this answers is not in the document]
Comment (MarianDiakiv) [reply service/src/test/java/greencity/service/HabitServiceImplTest.java]: done

**Review by airetOK — APPROVED** [review]

**Review by forestik — APPROVED** [review]

**Review by Skajl-dev — APPROVED** [review]

**Pull request merged**